include/boost/beast/experimental/core/impl/url_parser.ipp

```diff
@@ -0,0 +1,352 @@
+//
+// Copyright (c) 2018 jackarain (jack dot wgm at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#include <boost/beast/experimental/core/url_parser.hpp>
+
+namespace boost {
+namespace beast {
+
+//--------------------------------------------------------------------------
+// scheme:[//[user[:password]@]host[:port]][/path][?query][#fragment]
+
+using boost::beast::string_view;
+
+string_view url_parser::make_string_view(const char* b, std::size_t s)
+{
+    return string_view(b, s);
+}
+
+bool url_parser::isunreserved(const char c) const
+{
+    if (std::isalpha(c) || std::isdigit(c) ||
+        c == '-' || c == '.' || c == '_' || c == '~')
+        return true;
+    return false;
+}
+
+bool url_parser::isuchar(const char c) const
+{
+    if (isunreserved(c) || c == ';' || c == '?' || c == '&' || c == '=')
+        return true;
+    return false;
+}
+
+bool url_parser::ishsegment(const char c) const
+{
+    if (isuchar(c) ||
+        c == ';' || c == ':' || c == '@' || c == '&' || c == '=')
+        return true;
+    return false;
+}
+
+bool url_parser::issubdelims(const char c) const
+{
+    if (c == '!' || c == '$' || c == '\'' || c == '(' ||
+        c == ')' || c == '*' || c == '+' || c == ',' ||
+        c == '=')
+        return true;
+    return false;
+}
+
+//--------------------------------------------------------------------------
+
+bool url_parser::parse(const std::string& url)
```

Comment:
eventually this will need to use error codes but we can ship the first experimental version using plain bool.

Reply:
Alright, I'll look at other error code implementations in preparation. I thought the bool return was coarse for all the error conditions.